### PR TITLE
Report Details View

### PR DESCRIPTION
**Pull request by marykatefain**
I apologize in advance for the size of this PR. Related to #32 
This PR introduces one new major feature: Report Details View, as well as a couple of auxiliary features.

### 1. Report Details View
- New view accessible from the Mod Report Queues: `/mod/reports/details/<sub>/<type>/<id>`
- This report details view contains all the basics about a given report, and allows the mods to complete basic tasks from the report view rather than having to go all over the site to do Mod Things.
- Action items include: Close/Reopen Report, Close All Related Reports, Delete Comment/Post, Ban user from sub, and (if admin), Ban user from the site
- "Close All Related Reports" is the only thing that is really a new feature here. The rest were already possible, but you had to run all around the site to do them. "Close All Related Reports" introduces a new param to the getReports() function, `related=True`, which returns all reports on a given post or comment

Screenshot from 2020-07-13 17 06 58

### 2. Un-ban User from Site
- Admins may now unban a banned user from the user profile

Screenshot from 2020-07-13 16 53 04

### 3. Mod Dashboard UI improvement
- Just a very slight improvement to the Mod Dashboard from how it previously was

Screenshot from 2020-07-13 16 28 27

### Some Notes:
- Post previews only work for text posts currently. Previews for link or upload posts just say "preview could not be rendered". I think this could come later.
- When a post is deleted by a Mod or Admin, it can still be viewed in the Report Details View. If it was deleted by the user, however, it can not. In a future PR I may add more nuance to this, such as allowing Admin to see self-deleted posts.
- I started this PR before the Weezy escape HTML drama, so please check that everything has been escaped correctly!

### CODE REVIEW

File: app/views/do.py [hunk 1]
```diff
@@ -2363,3 +2391,96 @@ def close_comment_report(id, action):
...
+        except SubPost.DoesNotExist:
+            error = jsonify(status='error', error=_('Post does not exist'))
+
+        try:
+            sub = Sub.get(Sub.sid == post.sid)
+        except Sub.DoesNotExist:
+            error = jsonify(status='error', error=_('Sub does not exist'))
+
+        if not current_user.is_mod(sub.sid) and not current_user.is_admin():
+            error = jsonify(status='error', error=[_('Not authorized')])
```
Comment (Polsaker) [inline app/views/do.py:2419]: You're returning string errors in this function except in this line where it's an array

File: app/views/do.py [hunk 2]
```diff
@@ -2363,3 +2391,96 @@ def close_comment_report(id, action):
...
```
Comment (Polsaker) [inline app/views/do.py:2468]: I spy another array

File: app/misc.py [hunk 3]
```diff
@@ -913,11 +913,11 @@ def getPostList(baseQuery, sort, page):
             posts = baseQuery.order_by(hot.desc()).limit(100).paginate(page, 25)
         elif config.database.engine == 'SqliteDatabase':
             posts = baseQuery.order_by(
-                (SubPost.score * 20 + (fn.datetime(SubPost.posted, 'unixepoch') - 1134028003) / 1500).desc()).limit(
+                (SubPost.score * 20 + (fn.Unix_Timestamp(SubPost.posted) - 1134028003) / 1500).desc()).limit(
                 100).paginate(page, 25)
         else:
             posts = baseQuery.order_by(
-                (SubPost.score * 20 + (fn.Unix_Timestamp(SubPost.posted) - 1134028003) / 1500).desc()).limit(
+                (SubPost.score * 20 + (fn.datetime(SubPost.posted, 'unixepoch') - 1134028003) / 1500).desc()).limit(
```
Comment (Polsaker) [inline app/misc.py:920]: This is breaking mysql support

Comment (marykatefain) [reply app/misc.py:920]: oops didn't mean to commit that one - I have to do this every time to make it work locally for me lol :sweat_smile:

File: app/html/mod/reportdetails.html [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+                <button class="sbm-post pure-button pure-button-primary banuserbutton">@{_('Ban')} @{report['reported']} @{_('from')} @{config.site.sub_prefix}/@{report['sub']}</button>
+            @end
+            @if current_user.is_admin() and reported_user.status != 5:
+                <form method="POST" data-reload="true" id="banuser" action="@{url_for('do.ban_user', username=report['reported'])}">
+                    @{form.DummyForm().csrf_token()!!html}
+                    <button href="javascript:{}" onclick="if(confirm('@{_('Are you sure you want to ban this user?')}')){document.getElementById('banuser').submit();}" class="sbm-post pure-button button-secondary">@{_('Ban')} @{report['reported']} @{_('from')} @{config.site.name}</button>
+                </form>
+            @end
+            @if reported_user.status == 5:
+              <button class="sbm-post pure-button pure-button-primary disabled">@{report['reported']} @{_('has been banned from')} @{config.site.name}</button>
```
[the remark this answers is not in the document]
Comment (marykatefain) [reply app/html/mod/reportdetails.html:101]: oyy I also just realized Admins can ban themselves? oops

**Pull request merged**